icu4c/source/common/locid.cpp

```diff
@@ -477,6 +483,30 @@ Locale &Locale::operator=(const Locale &other)
     return *this;
 }
 
+Locale& Locale::operator=(Locale&& other) U_NOEXCEPT {
+    setToBogus();
+
+    if (other.fullName == other.fullNameBuffer) {
+        uprv_strcpy(fullNameBuffer, other.fullNameBuffer);
+    } else {
+        fullName = other.fullName;
+    }
+
+    baseName = other.baseName == other.fullName ? fullName : other.baseName;
+
+    uprv_strcpy(language, other.language);
+    uprv_strcpy(script, other.script);
+    uprv_strcpy(country, other.country);
+
+    variantBegin = other.variantBegin;
+    fIsBogus = other.fIsBogus;
+
+    other.baseName = other.fullName = other.fullNameBuffer;
+    other.fIsBogus = TRUE;
```

Comment:
Please replace this with `other.variantBegin = 0;`

Reply:
Oh, and `*other.fullNameBuffer = 0;`  right?

Reply:
Just call `other.setToBogus()`.  Everything you need to do here is covered by that helper function, and not much more.